### PR TITLE
Fix fetching GitLab snapshots on repos with no tags

### LINKED ISSUE #242: GitLab with `--version=branch` fails if the repository has no tags
With this Nix expression:

<details>
<summary>Details</summary>

```nix
{
  lib,
  stdenv,
  fetchFromGitLab,
  libdecor,
  libinput,
  libxkbcommon,
  meson,
  ninja,
  pixman,
  pkg-config,
  systemd,
  wayland,
  wayland-protocols,
  wlroots_0_16,
  xorg,
  nix-update-script,
}:

stdenv.mkDerivation {
  pname = "flyaway";
  version = "unstable-2023-07-04";

  src = fetchFromGitLab {
    owner = "serebit";
    repo = "flyaway";
    rev = "dc2ce34f54fc0485e39783b6e9c28dd624d61544";
    hash = "sha256-SSMe1RdMoAyOzPTQrH5X0P5JZK7FlRxyysmkHkDoFuM=";
  };

  nativeBuildInputs = [
    meson
    ninja
    pkg-config
  ];

  buildInputs = [
    libdecor
    libinput
    libxkbcommon
    pixman
    systemd
    wayland
    wayland-protocols
    wlroots_0_16
    xorg.libxcb
    xorg.xcbutilwm
  ];

  passthru.updateScript = nix-update-script {
    extraArgs = [
      "--version"
      "branch"
    ];
  };

  meta = {
    mainProgram = "flyaway";
    description = "A test compositor to gain familiarity with Wayland and wlroots";
    homepage = "https://gitlab.com/serebit/flyaway";
    license = lib.licenses.asl20;
    platforms = lib.platforms.linux;
    maintainers = with lib.maintainers; [ federicoschonborn ];
  };
}
```

</details>

`nix-update -F flyaway --version branch` (and `branch=main`) fails since there are no tags made in the [repository](https://gitlab.com/serebit/flyaway).

The GitLab snapshot fetcher tries to find a release tag to append the `unstable-...` bit here:

https://github.com/Mic92/nix-update/blob/5a5f48412d48e336da87eff0bafe67419992db94/nix_update/version/gitlab.py#L55-L56

But the release fetcher raises an error if there are no tags:

https://github.com/Mic92/nix-update/blob/5a5f48412d48e336da87eff0bafe67419992db94/nix_update/version/gitlab.py#L26-L27

### PR DESCRIPTION
**Pull request by FedericoSchonborn**
This commit fixes #242

### CODE REVIEW

File: nix_update/version/gitlab.py [hunk 1]
```diff
@@ -52,7 +52,10 @@ def fetch_gitlab_snapshots(url: ParseResult, branch: str) -> list[Version]:
     resp = urllib.request.urlopen(gitlab_url)
     commits = json.load(resp)
 
-    versions = fetch_gitlab_versions(url)
+    try:
+        versions = fetch_gitlab_versions(url)
+    except VersionError:
+        versions = []
     latest_version = versions[0].number if versions else "0"
```
Comment (Mic92) [inline nix_update/version/gitlab.py:59]: I think if we don't have any versions than it should be `unstable-{date}` rather than `0-unstable-{date}`.

Comment (FedericoSchonborn) [reply nix_update/version/gitlab.py:59]: Isn't `0-unstable-{date}` recommended by the Nixpkgs docs? AFAIK the GitHub fetchers do the same.

**Comment by Mic92** [comment]
@mergify queue

**Comment by Mic92** [comment]
Test errors look unrelated. Will fix them now.

**Pull request merged**